### PR TITLE
SQUASH ME: adapt uart test

### PR DESCRIPTION
**Pull request by OlegHahm**

### CODE REVIEW

File: tests/periph_uart/main.c [hunk 1]
```diff
@@ -51,12 +51,12 @@ static char printer_stack[THREAD_STACKSIZE_MAIN];
 
 static int parse_dev(char *arg)
 {
-    int dev = atoi(arg);
+    unsigned dev = atoi(arg);
     if (dev == UART_STDIO_DEV) {
```
Comment (gebart) [inline tests/periph_uart/main.c]: `if (UART_DEV(dev) == UART_STDIO_DEV)`

File: tests/periph_uart/main.c [hunk 2]
```diff
@@ -51,12 +51,12 @@ static char printer_stack[THREAD_STACKSIZE_MAIN];
...
         printf("Error: The selected UART_DEV(%i) is used for the shell!\n", dev);
         return -2;
     }
-    if (dev < 0 || (uart_t) dev >= UART_NUMOF) {
+    if ((uart_t) dev >= UART_NUMOF) {
```
Comment (gebart) [inline tests/periph_uart/main.c]: don't cast to uart_t

File: tests/periph_uart/main.c [hunk 3]
```diff
@@ -51,12 +51,12 @@ static char printer_stack[THREAD_STACKSIZE_MAIN];
...
         printf("Error: Invalid UART_DEV device specified (%i).\n", dev);
```
Comment (gebart) [inline tests/periph_uart/main.c]: %u now

**Comment by haukepetersen** [comment]
closed in favor of slightly different fix

**Pull request closed without merging**